docs/cookbook/makeExpandedTemporal.mjs

```diff
@@ -0,0 +1,175 @@
+// number of digits.
+function formatExpandedYear(year) {
+  let yearString;
+  if (year < 1000 || year > 9999) {
+    let sign = year < 0 ? '-' : '+';
+    let yearNumber = bigIntAbs(year);
+    yearString = sign + `${yearNumber}`.padStart(10, '0');
+  } else {
+    yearString = `${year}`;
+  }
+  return yearString;
+}
+
+function isLeapYear(year) {
+  const isDiv4 = year % 4n === 0n;
+  const isDiv100 = year % 100n === 0n;
+  const isDiv400 = year % 400n === 0n;
+  return isDiv4 && (!isDiv100 || isDiv400);
+}
+
+// This checks to see if the ISO string matches our 10-digit expanded year
+// format, and if so, returns both the expanded year as a BigInt, and a new
+// ISO string with an in-range year that can be passed to the original
+// Temporal string parsing functions.
+// The in-range year is 1972 if the expanded year is a leap year, and
+// otherwise 1970, so that the rules for February 29 remain correct.
+// See the note about the number of digits in formatExpandedYear().
+function parseExpandedYear(isoString) {
+  const matchExpandedYear = /^[-+\u2212]\d{10}/;
+  const result = matchExpandedYear.exec(isoString);
+  if (!result) return { isoString };
+  const expandedYear = BigInt(result[0]);
+  const isoYear = isLeapYear(expandedYear) ? 1972 : 1970;
+  return {
+    expandedYear,
+    isoString: isoString.replace(matchExpandedYear, isoYear.toString())
+  };
+}
+
+// This is a map of Temporal objects to their expanded year (as BigInt).
+// The data model consists of the Temporal object (with the ISO year set
+// internally to 1970 or 1972) and the expanded year. This map is used to
+// associate Temporal objects with their expanded years, instead of defining
+// extra properties on the Temporal object.
+const expandedYears = new WeakMap();
+
+class ExpandedPlainDate extends Temporal.PlainDate {
+  // The expanded-year versions of the Temporal types are limited to using the
+  // ISO calendar.
+  constructor(year, isoMonth, isoDay) {
+    year = BigInt(year);
+    const isoYear = isLeapYear(year) ? 1972 : 1970;
+    super(isoYear, isoMonth, isoDay, 'iso8601');
+    expandedYears.set(this, year);
+  }
+
+  static _convert(plainDate, expandedYear) {
+    if (plainDate instanceof ExpandedPlainDate) return plainDate;
+    const f = plainDate.getISOFields();
+    return new this(expandedYear, f.isoMonth, f.isoDay);
+  }
+
+  static from(item) {
+    if (typeof item === 'string') {
+      const { expandedYear, isoString } = parseExpandedYear(item);
+      item = Temporal.PlainDate.from(isoString);
+      if (expandedYear) return this._convert(item, expandedYear);
+    }
+    if (item instanceof Temporal.PlainDate) {
+      return this._convert(item, BigInt(item.year));
+    }
+    const calendar = Temporal.Calendar.from('iso8601');
+    return calendar.dateFromFields(item, undefined, this);
+  }
+
+  // This overrides the .year property to return the expanded year instead. It
+  // could also be done with a calendar instead.
```

Comment:
I think It couldn't be done with a calendar instead because otherwise year() would check that what the calendar returns is an integer, not a bigint. That's why it's necessary to override, and only works for code where the users aren't expecting temporal to return ints here.

Reply:
Good point. I guess if you were doing this with a calendar, you'd need to add a separate `expandedYear` field.